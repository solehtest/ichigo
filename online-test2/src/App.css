/* desktop screen */
.item1 { 
  grid-area: one; 
  background-color: royalblue;
}
.item2 { 
  grid-area: two; 
  background-color: crimson;
}
.item3 { 
  grid-area: three;
  background-color: peru; 
}
.item4 { 
  grid-area: four; 
  background-color: indianred;
}
.item5 { 
  grid-area: five; 
  background-color: purple;
}
.item6 { 
  grid-area: six; 
  background-color: palegreen;
}
.item7 { 
  grid-area: seven; 
  background-color: teal;
}
.item8 { 
  grid-area: eight; 
  background-color: mediumslateblue;
}
.item9 { 
  grid-area: nine; 
  background-color: darkslategrey;
}

.grid-container {
  color: white;
  display: grid;
  grid-template-areas:
    'one one two two two two'
    'one one three three four four'
    'five seven seven seven seven eight'
    'six seven seven seven seven nine';
  gap: 10px;
  background-color: whitesmoke;
  padding: 10px;
}

.grid-container > div {
  text-align: center;
  padding: 20px 0;
  font-size: 30px;
}

/* mobile screen */
.item-small1 { 
  grid-area: one; 
  background-color: royalblue;
}
.item-small2 { 
  grid-area: two; 
  background-color: crimson;
}
.item-small3 { 
  grid-area: three;
  background-color: peru; 
}
.item-small4 { 
  grid-area: four; 
  background-color: indianred;
}
.item-small5 { 
  grid-area: five; 
  background-color: purple;
}
.item-small6 { 
  grid-area: six; 
  background-color: palegreen;
}
.item-small7 { 
  grid-area: seven; 
  background-color: teal;
}
.item-small8 { 
  grid-area: eight; 
  background-color: mediumslateblue;
}
.item-small9 { 
  grid-area: nine; 
  background-color: darkslategrey;
}

.grid-container-small {
  color: white;
  display: grid;
  grid-template-areas:
    'one one'
    'three four'
    'two two'
    'seven seven'
    'five eight'
    'six nine';
  gap: 10px;
  background-color: whitesmoke;
  padding: 10px;
}

.grid-container-small > div {
  text-align: center;
  padding: 20px 0;
  font-size: 30px;
}