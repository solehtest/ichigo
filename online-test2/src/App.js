import './App.css';
import MediaQuery from 'react-responsive'

function App() {
  // list of colors
  const colors = ['royalblue', 'crimson', 'peru', 'indianred', 
                  'purple', 'palegreen', 'teal', 'mediumslateblue',
                  'darkslategrey'];

  // shuffle color in fullscreen mode
  const handleClickFullscreen = () => {
    let shuffleColors = colors.sort(() => Math.random() - 0.5);
    for (let i = 0; i < shuffleColors.length; i++) {
      document.getElementsByClassName('item'+(i+1))[0].style.backgroundColor = shuffleColors[i];
    }
  };

  // shuffle color in mobile mode
  const handleClickMobile = () => {
    let shuffleColors = colors.sort(() => Math.random() - 0.5);
    for (let i = 0; i < shuffleColors.length; i++) {
      document.getElementsByClassName('item-small'+(i+1))[0].style.backgroundColor = shuffleColors[i];
    }
  }

  return (
    <div>
      <MediaQuery minWidth={601}>
        <div className="grid-container">
          <div className="item1" onClick={handleClickFullscreen}>1</div>
          <div className="item2" onClick={handleClickFullscreen}>2</div>  
          <div className="item3" onClick={handleClickFullscreen}>3</div>
          <div className="item4" onClick={handleClickFullscreen}>4</div>
          <div className="item5" onClick={handleClickFullscreen}>5</div>
          <div className="item6" onClick={handleClickFullscreen}>6</div>
          <div className="item7" onClick={handleClickFullscreen}>7</div>
          <div className="item8" onClick={handleClickFullscreen}>8</div>
          <div className="item9" onClick={handleClickFullscreen}>9</div>
        </div>
      </MediaQuery>
      <MediaQuery maxWidth={600}>
        <div className="grid-container-small">
          <div className="item-small1" onClick={handleClickMobile}>1</div>
          <div className="item-small3" onClick={handleClickMobile}>3</div>  
          <div className="item-small4" onClick={handleClickMobile}>4</div>
          <div className="item-small2" onClick={handleClickMobile}>2</div>
          <div className="item-small7" onClick={handleClickMobile}>7</div>
          <div className="item-small5" onClick={handleClickMobile}>5</div>
          <div className="item-small8" onClick={handleClickMobile}>8</div>
          <div className="item-small6" onClick={handleClickMobile}>6</div>
          <div className="item-small9" onClick={handleClickMobile}>9</div>
        </div>
      </MediaQuery>
    </div>
  );
}

export default App;
