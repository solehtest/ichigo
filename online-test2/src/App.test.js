import { render, screen } from '@testing-library/react';
import App from './App';
import { Context as ResponsiveContext } from 'react-responsive'

test('renders as fullscreen', () => {
  const { container: layout } = render(
    <ResponsiveContext.Provider value={{ width: 601 }}>
      <App />
    </ResponsiveContext.Provider>
  )
  expect(layout).toMatchSnapshot()

  let containers = layout.getElementsByClassName('grid-container')[0].childNodes;
  let values = [];
  for (let i = 0; i < containers.length; i++) {
    values.push(containers[i].textContent);
  }
  console.log('Fullscreen Box : ' + values)

  // text 1
  let elementOne = screen.getByText(values[0]);
  expect(elementOne).toBeInTheDocument();
  expect(elementOne).toHaveClass('item1');
  
  // text 2
  let elementTwo = screen.getByText(values[1]);
  expect(elementTwo).toBeInTheDocument();
  expect(elementTwo).toHaveClass('item2');

  // text 3
  let elementThree = screen.getByText(values[2]);
  expect(elementThree).toBeInTheDocument();
  expect(elementThree).toHaveClass('item3');

  // text 4
  let elementFour = screen.getByText(values[3]);
  expect(elementFour).toBeInTheDocument();
  expect(elementFour).toHaveClass('item4');

  // text 5
  let elementFive = screen.getByText(values[4]);
  expect(elementFive).toBeInTheDocument();
  expect(elementFive).toHaveClass('item5');

  // text 6
  let elementSix = screen.getByText(values[5]);
  expect(elementSix).toBeInTheDocument();
  expect(elementSix).toHaveClass('item6');

  // text 7
  let elementSeven = screen.getByText(values[6]);
  expect(elementSeven).toBeInTheDocument();
  expect(elementSeven).toHaveClass('item7');

  // text 8
  let elementEight = screen.getByText(values[7]);
  expect(elementEight).toBeInTheDocument();
  expect(elementEight).toHaveClass('item8');

  // text 9
  let elementNine = screen.getByText(values[8]);
  expect(elementNine).toBeInTheDocument();
  expect(elementNine).toHaveClass('item9');
});

test('renders as mobile', () => {
  const { container: layout } = render(
    <ResponsiveContext.Provider value={{ width: 300 }}>
      <App />
    </ResponsiveContext.Provider>
  )
  expect(layout).toMatchSnapshot()

  let containers = layout.getElementsByClassName('grid-container-small')[0].childNodes;
  let values = [];
  for (let i = 0; i < containers.length; i++) {
    values.push(containers[i].textContent);
  }
  console.log('Mobile Box : ' + values)

  // text 1
  let elementOne = screen.getByText(values[0]);
  expect(elementOne).toBeInTheDocument();
  expect(elementOne).toHaveClass('item-small1');

  // text 3
  let elementThree = screen.getByText(values[1]);
  expect(elementThree).toBeInTheDocument();
  expect(elementThree).toHaveClass('item-small3');

  // text 4
  let elementFour = screen.getByText(values[2]);
  expect(elementFour).toBeInTheDocument();
  expect(elementFour).toHaveClass('item-small4');

  // text 2
  let elementTwo = screen.getByText(values[3]);
  expect(elementTwo).toBeInTheDocument();
  expect(elementTwo).toHaveClass('item-small2');

  // text 7
  let elementSeven = screen.getByText(values[4]);
  expect(elementSeven).toBeInTheDocument();
  expect(elementSeven).toHaveClass('item-small7');

  // text 5
  let elementFive = screen.getByText(values[5]);
  expect(elementFive).toBeInTheDocument();
  expect(elementFive).toHaveClass('item-small5');

  // text 8
  let elementEight = screen.getByText(values[6]);
  expect(elementEight).toBeInTheDocument();
  expect(elementEight).toHaveClass('item-small8');

  // text 6
  let elementSix = screen.getByText(values[7]);
  expect(elementSix).toBeInTheDocument();
  expect(elementSix).toHaveClass('item-small6');

  // text 9
  let elementNine = screen.getByText(values[8]);
  expect(elementNine).toBeInTheDocument();
  expect(elementNine).toHaveClass('item-small9');
});