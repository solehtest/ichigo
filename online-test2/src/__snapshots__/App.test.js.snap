// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as fullscreen 1`] = `
<div>
  <div>
    <div
      class="grid-container"
    >
      <div
        class="item1"
      >
        1
      </div>
      <div
        class="item2"
      >
        2
      </div>
      <div
        class="item3"
      >
        3
      </div>
      <div
        class="item4"
      >
        4
      </div>
      <div
        class="item5"
      >
        5
      </div>
      <div
        class="item6"
      >
        6
      </div>
      <div
        class="item7"
      >
        7
      </div>
      <div
        class="item8"
      >
        8
      </div>
      <div
        class="item9"
      >
        9
      </div>
    </div>
  </div>
</div>
`;

exports[`renders as mobile 1`] = `
<div>
  <div>
    <div
      class="grid-container-small"
    >
      <div
        class="item-small1"
      >
        1
      </div>
      <div
        class="item-small3"
      >
        3
      </div>
      <div
        class="item-small4"
      >
        4
      </div>
      <div
        class="item-small2"
      >
        2
      </div>
      <div
        class="item-small7"
      >
        7
      </div>
      <div
        class="item-small5"
      >
        5
      </div>
      <div
        class="item-small8"
      >
        8
      </div>
      <div
        class="item-small6"
      >
        6
      </div>
      <div
        class="item-small9"
      >
        9
      </div>
    </div>
  </div>
</div>
`;
